### ISSUE TITLE
Rule Change: `func-style` ignore overloaded function declarations

### ISSUE THREAD

**Issue by JoshuaKGoldberg** [eslint/eslint#19738]
### What rule do you want to change?

func-style

### What change do you want to make?

Generate fewer warnings

### How do you think the change should be implemented?

A new default behavior

### Example code

```js
/* eslint func-style: "error" */

// func-style report: Expected a function expression.
function noOverloads() {}

// No func-style report
function hasOverloads(value: string): string
function hasOverloads(value: number): number
function hasOverloads(value: unknown) {
  return value;
}
```

### What does the rule currently do for this code?

```plaintext
Expected a function expression. 9:1 - 11:2
```

### What will the rule do after it's changed?

Not report on the function declaration.

### Participation

- [x] I am willing to submit a pull request to implement this change.

### Additional comments

Quoting https://github.com/eslint/eslint/issues/19719#issuecomment-2880017202:

> I would take [JoshuaKGoldberg's suggestion](https://github.com/eslint/eslint/issues/19719#issuecomment-2876875662) a step further and always allow overloaded function declarations. Don’t even add the option to report it. Otherwise it’s simply a TypeScript error.
> 
> [Playground](https://www.typescriptlang.org/play/?#code/GYVwdgxgLglg9mABACwIYGcDyA3ApgJwBs5UATdACm1UJFwC5F0p8YwBzASkedY4FgAUKEiwEKDDgLEylarQaIwIALYAjAtyWqN+IdXwSseIiXKIAvIhHR4YKjTqNwAazBwA7mE6IA3kMREfFwoEHwkeToAbiEAXyA)
> 
> ```ts
> function hasOverloads(value: string): string
> function hasOverloads(value: number): number
> var hasOverloads = function(value: unknown) {
>   return value;
> }
> ```
>
> If there’s user demand, there could be a rule to disallow TypeScript function overloads. But IMO that feels out of scope for `func-style`.

**Comment by nzakas** [comment]
This seems like a good suggestion to me, so I'm :+1:. Would like to hear some other thoughts from @eslint/eslint-team

**Comment by Tanujkanti4441** [comment]
I am in favor of this change.

**Comment by mdjermanovic** [comment]
Makes sense to me, marking as accepted.
